sdk/web-pubsub/web-pubsub-client/src/webPubSubClient.ts

```diff
@@ -686,35 +686,41 @@ export class WebPubSubClient {
           throw err;
         }
 
-        try {
-          switch (message.kind) {
-            case "ack": {
-              handleAckMessage(message as AckMessage);
-              break;
-            }
-            case "connected": {
-              handleConnectedMessage(message as ConnectedMessage);
-              break;
-            }
-            case "disconnected": {
-              handleDisconnectedMessage(message as DisconnectedMessage);
-              break;
-            }
-            case "groupData": {
-              handleGroupDataMessage(message as GroupDataMessage);
-              break;
-            }
-            case "serverData": {
-              handleServerDataMessage(message as ServerDataMessage);
-              break;
+        if (!Array.isArray(messages)) {
+          messages = [messages];
+        }
+
+        messages.forEach((message) => {
```

Comment:
style nit: for/of is a little more readable and saves you from creating a function scope

```suggestion
        for (const message of messages) {
```